### DISCUSSION TITLE
new folder in extensions

### DISCUSSION THREAD

**Discussion by smohamadabedy** [opencart/opencart#10975] [Ideas]
hi congratulations to the team and all contributors for new version, actually dedicating a folder for extension is helpful, but limiting folders to be in one of predefined is annoying, how about letting group extensions have their own folder and extension controller , model  and language like (module group , payment or theme ) 
As there is no official documentation about extensions yet I couldn't figure out if this is implemented

**Comment by felipoantonoff** [comment]
Hi @smohamadabedy I imagine that its folders and subfolders of the modules can be defined.
Example:
/extension/your_brand/ -> admin, catalog and etc and it seems to have an install.json that informs some details

Now I also don't know if it can stay:
/extension/your_marca/modulo_x/ -> admin, catalog and etc, which I believe to be ideal maybe or if not each module would be /extension/modulo_x.

I also found it very good for organization and easier access to changes.
When documentation comes, it may be clearer for everyone.

What would be your idea, I couldn't understand very well?
/extension/your_brand/module_x/ -> admin, catalog and etc, which I think is ideal, maybe or if not, each module would be /extension/modulo_x or all of them inside /extension/your_brand/ -> admin/catalog and etc as it is today Opencart natives, well anyway it's better than other places for organization and easier access for changes.

When documentation comes, it may be clearer for everyone.

What would be your idea, I couldn't understand very well?

**Reply by smohamadabedy** [reply]
well to clarify I want something like this after installation:
extension/admin/constroller/mybrand/myfile.php
extension/admin/language/mybrand/myfile.php
extension/admin/model/mybrand/myfile.php

**Reply by felipoantonoff** [reply]
I understand you want to keep the same base path as before, but putting it in the /extension folder, it's an idea so you don't need to migrate the path of all modules if that's it.

But due to the install I don't think it will get this pattern, but if it does, great too :)
